### PR TITLE
Sprite updates & state fix for zippo lighters

### LINKED ISSUE #21257: Zippos don't have an in-hand sprite when turned on
**Issue Description**:
Zippos don't have an in-hand sprite when they are turned on.

**What did you expect to happen**: 
To at least keep the display of the zippo, instead of it just simply vanishing.

**What happened instead**:
image
image

**Why is this bad/What are the consequences**:
Just a small sprite bug.

**Steps to reproduce the problem**:
1. Get zippo
2. Turn it on
3. Magic zippo disappears

**When did the problem start happening**:
Unsure

### PR DESCRIPTION
**Pull request by Stealthii**
## What Does This PR Do
This PR adds two new methods: `update_icon_state()` and `update_overlays()`, which handle sprite updates for held and UI lighter items, and drops the use of explicit 'icon_state' definitions for lit/unlit in favour of consistently named sprites for lighters.

The held lighter items have been split into a separate file - `lighter_lefthand.dmi` and `lighter_righthand.dmi` by @XFirebirdX, and new sprites for the colored and lit versions of zippos, which fixes #21257.

The naming of zippo sprites has been updated to reflect this change and stay consistent with the base lighter object and sprites.

## Why It's Good For The Game
Fixes a long standing bug, and brings lighter code more in line with changes introduced in **Port TG updating appearances (#17943)**, as well as adding more flair to the use of zippo lighters.

## Images of changes
| Held form | Image |
| --- | --- |
| Base (silver) | base |
| NT (gold) | gold |
| Contractor (black) | black |
| Gonzo (red) | red |
| Captain (blue) | blue |
| @warriorstar-orion (purple) | purple-b |

https://github.com/ParadiseSS13/Paradise/assets/998920/61e7037f-0267-4517-9a9c-c8dc882252a8

## Testing
* Spawn in
* Give yourself every lighter in the game
* Light, unlight, and move lighters between hands to confirm that:
  * colored "held zippo" sprite exists for both lit/unlit states of all zippo lighters (for bugfix)
  * item lit/unlit sprites in HUD are unchanged and work correctly (for routine changes)

## Changelog
:cl:
fix: Base and custom zippo lighters no longer disappear from hand once lit
imageadd: Colored and lit variations of held zippo lighters
/:cl:

### CODE REVIEW

**Comment by Stealthii** [comment]
Closing this until ready for review - working with @XFirebirdX on missing sprites for the held versions!

**Comment by Henri215** [comment]
image

Ended up touching this dress by mistake?

**Comment by Stealthii** [comment]
Good catch @Henri215 - turns out the VSCode plugin for DMI has a bug in its pack handling which results in renaming having some side effects. Reapplied the naming change through Dream Maker and the sexy dress was unaffected.

**Review by Henri215 — COMMENTED** [review]
Only a few things left:
1. Remove the zippo in hand from the items_hand.dmi file since it is now in a different place

2. Update the PR title to better reflect the changes you are making (which is fixing zippos having no in hand sprites)

**Comment by Stealthii** [comment]
Alright - those changes have been made, and additionally updated the held sprite for @warriorstar-orion's custom zippo to have the correct blue flame.

File: code/modules/customitems/item_defines.dm [hunk 1]
```diff
@@ -202,37 +202,36 @@
 
 	to_chat(user, "<span class='warning'>You can't modify [target]!</span>")
 
+// Custom zippo lighters
+/obj/item/lighter/zippo/fluff
+	name = "custom zippo"
+	desc = "A custom made zippo lighter."
+	// HUD icons from custom_items
+	icon = 'icons/obj/custom_items.dmi'
```
Comment (Henri215) [inline code/modules/customitems/item_defines.dm:210]:
```suggestion
	desc = "A custom made zippo lighter."
	icon = 'icons/obj/custom_items.dmi'
```
That comment isnt necessary and we normally call it the item icon.
HUD icon makes it sound like its a sprite belonging to the icons you see while wearing HUD glasses.

**Review by Henri215 — APPROVED** [review]
LGTM

File: code/modules/customitems/item_defines.dm (hunk 1 above)
Comment (Stealthii) [reply code/modules/customitems/item_defines.dm]: Yeah I agree with you there. Made that change in ad1b5fb to reduce duplication, but the comment isn't necessary elsewhere and my word-fu isn't quite correct (as HUD and in-world items use the same icon)

**Review by S34NW — APPROVED** [review]

**Pull request merged**